### PR TITLE
Add ReloadAmmo trait and remove Rearm activity

### PR DESCRIPTION
**Pull request by reaperrr**
This is the first step towards getting rid of `Rearm`/`Repair` activities in favor of upgrade-based solutions for increased flexibility.

Allows actors to be rearmed and repaired at the same time.

This also moves the `RearmBuildings` property from `Aircraft` and `Minelayer` to `ReloadAmmo`.

Supercedes #9606.

### CODE REVIEW

**Comment by abcdefg30** [comment]
Needs a rebase already?

File: OpenRA.Mods.Common/AI/HackyAI.cs [hunk 1]
```diff
@@ -450,12 +450,15 @@ bool HasAdequateAirUnitReloadBuildings(ActorInfo actorInfo)
 			if (aircraftInfo == null)
 				return true;
 
-			var ammoPoolsInfo = actorInfo.TraitInfos<AmmoPoolInfo>();
+			// We assume that UpgradeMinEnabledLevel = 0 means the actor reloads ammo by itself.
+			// To have ReloadAmmo also implies having at least 1 AmmoPool.
+			var buildingReload = actorInfo.TraitInfos<ReloadAmmoInfo>()
+				.Where(x => x.UpgradeMinEnabledLevel > 0).ToList();
```
Comment (RoosterDragon) [inline OpenRA.Mods.Common/AI/HackyAI.cs]: You could use `.FirstOrDefault()` instead of `ToList()` since you only care about the first result.

**Comment by reaperrr** [comment]
Updated.

Now depends on #10630 to prevent helicopters from reloading while still in the air.

**Comment by reaperrr** [comment]
Updated.

**Comment by reaperrr** [comment]
By now I'm thinking about waiting for (or implementing it myself) some UpgradeDockedActor trait instead of using UpgradeActorsNear, so I'm closing this for now to clear the queue.

**Pull request closed without merging**